domain/src/test/kotlin/no/nav/su/se/bakover/domain/vedtak/VedtakTest.kt

```diff
@@ -0,0 +1,423 @@
+package no.nav.su.se.bakover.domain.vedtak
+
+import com.nhaarman.mockitokotlin2.mock
+import io.kotest.matchers.shouldBe
+import no.nav.su.se.bakover.common.Tidspunkt
+import no.nav.su.se.bakover.common.UUID30
+import no.nav.su.se.bakover.common.april
+import no.nav.su.se.bakover.common.februar
+import no.nav.su.se.bakover.common.januar
+import no.nav.su.se.bakover.common.mai
+import no.nav.su.se.bakover.common.mars
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.common.startOfDay
+import no.nav.su.se.bakover.domain.FnrGenerator
+import no.nav.su.se.bakover.domain.NavIdentBruker
+import no.nav.su.se.bakover.domain.Saksnummer
+import no.nav.su.se.bakover.domain.ValgtStønadsperiode
+import no.nav.su.se.bakover.domain.behandling.Attestering
+import no.nav.su.se.bakover.domain.behandling.Behandlingsinformasjon
+import no.nav.su.se.bakover.domain.grunnlag.Grunnlag
+import no.nav.su.se.bakover.domain.grunnlag.Grunnlagsdata
+import no.nav.su.se.bakover.domain.grunnlag.Uføregrad
+import no.nav.su.se.bakover.domain.søknadsbehandling.Søknadsbehandling
+import no.nav.su.se.bakover.domain.vilkår.Vilkårsvurderinger
+import org.junit.jupiter.api.Test
+import java.time.Clock
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+import java.util.UUID
+
+internal class VedtakTest {
+    private val fixedClock: Clock = Clock.fixed(1.januar(2021).startOfDay().instant, ZoneOffset.UTC)
+
+    @Test
+    fun `klarer å lage tidslinje av vedtaket`() {
+        //  1.jan       31.mars
+        //  |--u1---|-u2-|  uføre
+        //  |-----f1-----|  flykting
+        //  |------------|  Vedtaket
+
+        //          1.mars   30.april
+        //          |--u3----|  uføre
+        //          |--f2----|  flykting
+        //          |--------|  Vedtaket
+
+        //       1.feb             30.mai
+        //       |--u4---|-u5------|  uføre
+        //                            flykting
+        //       |-----------------|  Vedtaket
+
+        // |-------------| periode
+
+        // |--u1-|--u4--|
+        // |--f1-|
+
+        val søknadsbehandlingVedtak = Vedtak.fromSøknadsbehandling(
```

Comment:
Kunne kanskje gjort et forsøk på å redusere noe av boilerplaten for oppsett av et vedtak, slik at det ble litt enklere å få tak i de feltene som faktisk har noe å si i for testen.